src/commonMain/kotlin/io/github/detekt/sarif4k/SarifDataBindings.kt

```diff
@@ -339,13 +344,26 @@ data class Address (
  *
  * Key/value pairs that provide additional information about the version control details.
  */
-@Serializable
-data class PropertyBag (
+@Serializable(with = PropertyBag.Companion::class)
+@JvmInline
+value class PropertyBag(private val value: Map<String, Any?>) : Map<String, Any?> by value {
     /**
      * A set of distinct strings that provide additional information.
      */
-    val tags: List<String>? = null
-)
+    val tags: List<String>? get() = (value["tags"] as Collection<*>?)?.map { it as String }
```

Comment:
Related with https://github.com/detekt/sarif4k/pull/101#discussion_r1500334245

I'm not sure where to draw the line of validation here. I also thought about `?.map { it.toString() }` instead of `?.map { it as String }`. 

`?.map { it.toString() }` will not crash and will work nice with number tags. But it could create strange artifacts if inside `tags` you have another list, the `toString` of a list is its memory position. And if we don't crash someone could think that it is a correct tag. For that reson I think it is better to use the cast.

I did a change here, anyway, to improve the error messages.